spec/observables/defer-spec.ts

```diff
@@ -72,4 +72,16 @@ describe('Observable.defer', () => {
     expectObservable(e1, unsub).toBe(expected);
     expectSubscriptions(source.subscriptions).toBe(sourceSubs);
   });
+
+  type(() => {
```

Comment:
overloaded signature like combinelatest (#1189) might be few edge cases need these kind of test cases.